core/domain/html_cleaner_test.py

```diff
@@ -126,3 +126,57 @@ def test_strip_html_tags(self):
 
         for datum in test_data:
             self.assertEqual(html_cleaner.strip_html_tags(datum[0]), datum[1])
+
+
+class RteComponentExtractorUnitTests(test_utils.GenericTestBase):
+    '''Test the RTE component extractor.'''
+
+    def test_get_rte_components(self):
+        test_data = (
+            '<p>Test text&nbsp;'
+            '<oppia-noninteractive-math '
+            'raw_latex-with-value="&amp;quot;\\frac{x}{y}&amp;quot;">'
+            '</oppia-noninteractive-math></p><p>&nbsp;'
+            '<oppia-noninteractive-link '
+            'text-with-value="&amp;quot;Link&amp;quot;" '
+            'url-with-value="&amp;quot;https://www.example.com&amp;quot;">'
+            '</oppia-noninteractive-link>.</p>'
+            '<p>Video</p>'
+            '<p><oppia-noninteractive-video autoplay-with-value="false" '
+            'end-with-value="0" start-with-value="0" '
+            'video_id-with-value="&amp;quot;'
+            'https://www.youtube.com/watch?v=Ntcw0H0hwPU&amp;quot;">'
+            '</oppia-noninteractive-video><br></p>'
+        )
+
+        expected_components = [
+            {
+                'customization_args': {
+                    'text-with-value': u'&quot;Link&quot;',
```

Comment:
Sure. Do you think it's better to just unescape (in which case the value would be `u'"Link"'`) or to unescape and then evaluate (in which case the value would be `u'Link'`). I'm not sure how to evaluate without introducing security risks.